python_on_whales/components/service/cli_wrapper.py

```diff
@@ -173,6 +173,10 @@ def create(
         limit_memory: Optional[str] = None,
         limit_pids: Optional[int] = None,
         log_driver: Optional[str] = None,
+        network: Optional[str] = None,
+        restart_condition: Optional[str] = None,
+        restart_max_attempts: Optional[int] = None,
+        secrets: List[str] = [],
```

Comment:
For secret, reading the doc https://docs.docker.com/engine/reference/commandline/service_create/#secret I think it could be `Union[dict[str, str], str, None]` and we could craft the string `source=app-key,target=app,uid=1000,gid=1001,mode=0400` from a dict, that would avoid some string manipulation by the user. What do you think?

Reply:
Done! The changes are not exactly as you had proposed but let me know what do you think.